instructionAPI/h/Dereference.h

```diff
@@ -31,211 +31,83 @@
-      /// to be dereferenced and a type
-      /// indicating how the memory at the address in question is to be interpreted.
-      Dereference(Expression::Ptr addr, Result_Type result_type) : Expression(result_type), addressToDereference(addr)
-      {
-      }
-      virtual ~Dereference() 
-      {
-      }
-      /// A %Dereference has one child, which represents the address being dereferenced.
-      /// \param children Appends the child of this %Dereference to \c children.
-      virtual void getChildren(vector<InstructionAST::Ptr>& children) const
-      {
-	children.push_back(addressToDereference);
-	return;
-      }
-      virtual void getChildren(vector<Expression::Ptr>& children) const
-      {
-          children.push_back(addressToDereference);
-          return;
-      }
-      /// The use set of a %Dereference is the same as the use set of its children.
-      /// \param uses The use set of this %Dereference is inserted into \c uses.
-      virtual void getUses(set<InstructionAST::Ptr>& uses)
-      {
-	addressToDereference->getUses(uses);
-	return;
-      }
-        /// An %InstructionAST is used by a %Dereference if it is equivalent to the 
-        /// %Dereference or it is used by the lone child of the %Dereference
-        virtual bool isUsed(InstructionAST::Ptr findMe) const
-        {
-            return addressToDereference->isUsed(findMe) || *findMe == *this;
-        }
-
-      virtual std::string format(formatStyle) const
-      {
-    std::string retVal;
-#if defined(DEBUG_MEMORY_ACCESS_WIDTH)
-        switch(Expression::userSetValue.type)
-        {
-            case u8:
-                retVal += "u8 @ ";
-                break;
-            case s8:
-                retVal += "s8 @ ";
-                break;
-            case u16:
-                retVal += "u16 @ ";
-                break;
-            case s16:
-                retVal += "s16 @ ";
-                break;
-            case u32:
-                retVal += "u32 @ ";
-                break;
-            case s32:
-                retVal += "s32 @ ";
-                break;
-            case u64:
-                retVal += "u64 @ ";
-                break;
-            case s64:
-                retVal += "s64 @ ";
-                break;
-            case sp_float:
-                retVal += "float @ ";
-                break;
-            case dp_float:
-                retVal += "double @ ";
-                break;
-            case dbl128:
-                retVal += "packed double @ ";
-                break;
-            default:
-                retVal += "UNKNOWN SIZE @ ";
-                break;
-        }
-#endif
-    retVal += "[" + addressToDereference->format() + "]";
-        return retVal;
-      }
+namespace Dyninst { namespace InstructionAPI {
+  using std::vector;
 
-        virtual std::string format(Architecture arch, formatStyle) const
-        {
-#if defined(DEBUG_MEMORY_ACCESS_WIDTH)
```

Comment:
Are you sure these debug will never be used?

Reply:
These formatters are going away with Capstone. There are so many of these little debug macros everywhere that have clearly not been used since the original implementation. For example, this one doesn't cover all of the possible sizes *and* `Result` has its own formatter method.